commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -117,9 +111,35 @@ public static ThreadPoolExecutor newCachedThreadPool(ThreadFactory threadFactory
      * @throws NullPointerException if threadFactory is null
      */
     public static ThreadPoolExecutor newCachedThreadPool(ThreadFactory threadFactory, int threadTimeoutMillis) {
+        return newCachedThreadPool(threadFactory, threadTimeoutMillis, 0, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Creates a thread pool with a minimum of corePoolSize threads and a maximum of maximumPoolSize
+     * threads. Uses the provided ThreadFactory to create new threads when needed.
+     *
+     * @param threadFactory the factory to use when creating new threads
+     * @return the newly created thread pool
+     * @throws NullPointerException if threadFactory is null
+     */
+    public static ThreadPoolExecutor newCachedThreadPool(
+            ThreadFactory threadFactory, int corePoolSize, int maximumPoolSize) {
+        return newCachedThreadPool(threadFactory, DEFAULT_THREAD_POOL_TIMEOUT_MILLIS, corePoolSize, maximumPoolSize);
+    }
+
+    /**
+     * Creates a thread pool with a minimum of corePoolSize threads and a maximum of maximumPoolSize
+     * threads. Uses the provided ThreadFactory to create new threads when needed.
+     *
+     * @param threadFactory the factory to use when creating new threads
+     * @return the newly created thread pool
+     * @throws NullPointerException if threadFactory is null
+     */
+    public static ThreadPoolExecutor newCachedThreadPool(
+            ThreadFactory threadFactory, int threadTimeoutMillis, int corePoolSize, int maximumPoolSize) {
         return newThreadPoolExecutor(
-                0,
-                Integer.MAX_VALUE,
+                corePoolSize,
+                maximumPoolSize,
```

Comment:
@jnelson15 I don't think this will do what you desire as the backing queue is unbounded and won't return false for `offer`, so you'll never go above the min core pool thread count. see https://github.com/kimchy/kimchy.github.com/blob/master/_posts/2008-11-23-juc-executorservice-gotcha.textile

Reply:
Ah yeah good catch. So we'll want to use `LinkedBlockingQueue` here?

Reply:
Oh nevermind the blog says that's problematic too